packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -268,47 +269,382 @@ class _CupertinoAppState extends State<CupertinoApp> {
+        data: CupertinoUserInterfaceLevelData.base,
+        child: CupertinoTheme(
+          data: effectiveThemeData,
+          child: CupertinoSystemColors(
+            data: CupertinoSystemColors.fromSystem ?? CupertinoSystemColors.fallbackValues,
+            child: WidgetsApp(
+              key: GlobalObjectKey(this),
+              navigatorKey: widget.navigatorKey,
+              navigatorObservers: _navigatorObservers,
+              pageRouteBuilder: <T>(RouteSettings settings, WidgetBuilder builder) =>
+                CupertinoPageRoute<T>(settings: settings, builder: builder),
+              home: widget.home,
+              routes: widget.routes,
+              initialRoute: widget.initialRoute,
+              onGenerateRoute: widget.onGenerateRoute,
+              onUnknownRoute: widget.onUnknownRoute,
+              builder: widget.builder,
+              title: widget.title,
+              onGenerateTitle: widget.onGenerateTitle,
+              textStyle: effectiveThemeData.textTheme.textStyle,
+              color: widget.color ?? CupertinoColors.activeBlue,
+              locale: widget.locale,
+              localizationsDelegates: _localizationsDelegates,
+              localeResolutionCallback: widget.localeResolutionCallback,
+              localeListResolutionCallback: widget.localeListResolutionCallback,
+              supportedLocales: widget.supportedLocales,
+              showPerformanceOverlay: widget.showPerformanceOverlay,
+              checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
+              checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
+              showSemanticsDebugger: widget.showSemanticsDebugger,
+              debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
+              inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
+                return CupertinoButton.filled(
+                  child: const Icon(
+                    CupertinoIcons.search,
+                    size: 28.0,
+                    color: CupertinoColors.white,
+                  ),
+                  padding: EdgeInsets.zero,
+                  onPressed: onPressed,
+                );
+              },
+            ),
+          ),
         ),
       ),
     );
   }
 }
+
+/// Establishes a subtree where iOS system colors resolve to the given data.
+///
+/// Typically the given [CupertinoSystemColorsData] is resolved against its own
+/// [BuildContext] using [CupertinoSystemColorsData.resolveColors].
+class CupertinoSystemColors extends InheritedWidget {
+  /// Creates a widget that provides a given [CupertinoSystemColorsData] to its
+  /// descendants.
+  const CupertinoSystemColors({
+    Key key,
+    @required CupertinoSystemColorsData data,
+    Widget child,
+  }) : _data = data,
+       assert(data != null),
+       super(key: key, child: child);
+
+  /// Creates a widget that provides a [CupertinoSystemColorsData] extracted from
```

Comment:
We haven't provided this sort of thing for other themes (inherited widgets) and it's apt to be confusing as defined because the child will not "see" the CupertinoSystemColors. 

Prefer to leave this out now; write a utility in the test file that adds the extra context with a Builder. In the future, perhaps we'll update all themes to include such a factory.